### PR TITLE
Brewing Recipes

### PR DESCRIPTION
**Pull request by cinthesize**
Added initial support for brewing recipes, potential for more paper-only brewing-related stuff later.
For #2046

### CODE REVIEW

**Comment by acikek** [comment]
Regarding formatting:
- Put a space between `if`/`for` and its opening parenthesis, but don't do this with methods.
- Put a space between a closing parenthesis and an opening curly brace.

For example: `if(cond){` should be `if (cond) {`

File: plugin/src/main/java/com/denizenscript/denizen/objects/EnchantmentTag.java [hunk 1]
```diff
@@ -72,7 +73,7 @@ public static EnchantmentTag valueOf(String string, TagContext context) {
             ench = Enchantment.getByName(string.toUpperCase());
         }
         if (ench == null) {
-            ench = Enchantment.getByKey(new NamespacedKey("denizen", Utilities.cleanseNamespaceID(string)));
+            ench = Enchantment.getByKey(new NamespacedKey(Denizen.getInstance(), Utilities.cleanseNamespaceID(string)));
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/EnchantmentTag.java]: Thaaaat should probably be put back the way it was

File: plugin/src/main/java/com/denizenscript/denizen/utilities/AdvancedTextImpl.java [hunk 2]
```diff
@@ -105,4 +106,16 @@ public InventoryView openAnvil(Player player, Location loc) {
     public void teleportPlayerRelative(Player player, Location loc) {
         player.teleport(loc, PlayerTeleportEvent.TeleportCause.PLUGIN);
     }
+
+    public void registerBrewingRecipe(String keyName, ItemStack result, ItemStack[] inputItem, boolean inputExact, ItemStack[] ingredientItem, boolean ingredientExact){
+        throw new UnsupportedOperationException();
+    }
+
+    public void clearBrewingRecipes(){
+        throw new UnsupportedOperationException();
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/utilities/AdvancedTextImpl.java]: Please test on a Spigot server - this exception is going to fire and crash reloads.

Comment (cinthesize) [reply plugin/src/main/java/com/denizenscript/denizen/utilities/AdvancedTextImpl.java]: Ah, I left that in on accident, my intention was to just replace that with a return statement, or is there a better alternative to put there?

Comment (mcmonkey4eva) [reply plugin/src/main/java/com/denizenscript/denizen/utilities/AdvancedTextImpl.java]: just leave the method empty

Comment (cinthesize) [reply plugin/src/main/java/com/denizenscript/denizen/utilities/AdvancedTextImpl.java]: Gotcha

File: plugin/src/main/java/com/denizenscript/denizen/scripts/containers/core/ItemScriptHelper.java [hunk 3]
```diff
@@ -162,7 +164,7 @@ public static void registerShapedRecipe(ItemScriptContainer container, ItemStack
                 ingredients.add(items);
             }
         }
-        NamespacedKey key = new NamespacedKey("denizen", internalId);
+        NamespacedKey key = new NamespacedKey(Denizen.getInstance(), internalId);
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/scripts/containers/core/ItemScriptHelper.java]: this one should also be put back how it was

Comment (cinthesize) [reply plugin/src/main/java/com/denizenscript/denizen/scripts/containers/core/ItemScriptHelper.java]: Done

**Pull request merged**